### PR TITLE
feat: Symbolication of minidump events

### PR DESCRIPTION
**Pull request by jan-auer**
This is a followup on #6416 and enables frame symbolication without changing much of the current pipeline. CFI stackwalking is still missing, but the scanned stacktraces have sufficient quality.

See comments below for a couple of quirks.

**NOTE**: Since I changed some of the code determining final frames, we might regress with grouping somewhere.

screen shot 2017-10-25 at 15 59 30

### CODE REVIEW

File: src/sentry/lang/native/plugin.py [hunk 1]
```diff
@@ -26,6 +26,8 @@
 
 
 class NativeStacktraceProcessor(StacktraceProcessor):
+    supported_platforms = ('cocoa', 'native')
```
Comment (jan-auer) [inline src/sentry/lang/native/plugin.py]: We'll leave it this explicit for now and add the new `"native"` type wherever necessary. The way platforms are used now has to be changed anyway, so we can put more thought into this later.

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: src/sentry/lang/native/plugin.py [hunk 2]
```diff
@@ -155,7 +159,6 @@ def on_referenced(dsym_file):
         self.sym = Symbolizer(
             self.project,
             self.object_lookup,
-            arch=self.arch,
```
Comment (jan-auer) [inline src/sentry/lang/native/plugin.py]: The architecture is not needed here anymore. The object lookup will already resolve the correct architecture and symcaches are generated separately per architecture.

File: src/sentry/lang/native/plugin.py [hunk 3]
```diff
@@ -238,41 +248,45 @@ def process_frame(self, processable_frame, processing_task):
...
+                None
             if sfrm.get('lineno'):
                 new_frame['lineno'] = sfrm['lineno']
             if sfrm.get('colno'):
                 new_frame['colno'] = sfrm['colno']
-            if sfrm.get('package'):
-                new_frame['package'] = sfrm['package']
+            if sfrm.get('package') or processable_frame.data['obj'] is not None:
+                new_frame['package'] = sfrm.get(
+                    'package', processable_frame.data['obj'].name)
```
Comment (jan-auer) [inline src/sentry/lang/native/plugin.py]: We have to use the object's name here, as the `package` field is missing in events generated from minidumps. While the information would be present in Breakpad, we stopped to expose it in symbolic. In general, I also like the idea of reading it from the object more than having it in every frame.

File: src/sentry/models/minidump.py [hunk 4]
```diff
@@ -8,13 +8,21 @@
...
 from symbolic import ProcessState
 
 from sentry.constants import LOG_LEVELS_MAP
 from sentry.db.models import FlexibleForeignKey, Model, sane_repr
 from sentry.models.file import File
 
+_version_re = re.compile(r'(\d+\.\d+\.\d+)\s+(.*)')
+
+_minidump_os_mapping = {
+    'Mac OS X': 'macOS',
```
Comment (jan-auer) [inline src/sentry/models/minidump.py]: Breakpad uses a different naming schema for operating systems than what we have hardcoded in the symbolizer. For the sake of keeping this change small, I'm just mapping the names here. However, we should rather try not to use the os name in the pipeline. UI is a separate topic...

File: src/sentry/models/minidump.py [hunk 5]
```diff
@@ -67,10 +75,18 @@ def merge_minidump_event(data, minidump_path):
...
     device = context.setdefault('device', {})
-    os['name'] = info.os_name
-    os['version'] = info.os_version
+    os['type'] = 'os'  # Required by "get_sdk_from_event"
+    os['name'] = _minidump_os_mapping.get(info.os_name, info.os_name)
     device['arch'] = info.cpu_family
 
+    # Breakpad reports the version and build number always in one string,
+    # but a version number is guaranteed even on certain linux distros.
+    match = _version_re.search(info.os_version)
```
Comment (jan-auer) [inline src/sentry/models/minidump.py]: The version string reported by Breakpad starts with `x.y.z` but then continues with arbitrary garbage depending on the client. We just put it into `"build"` for now, but it doesn't really matter, actually.

 - **macOS**: build number
 - **Windows**: service pack and build number
 - **Linux**: output of `uname -a`

**Review by mitsuhiko — APPROVED** [review]
wfm

**Comment by mitsuhiko** [comment]
after mergin update getsentry uwsgi config and build deps in getsentry

**Pull request merged**